repair_stock_dest/models/repair.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2024 Patryk Pyczko (APSL-Nagarro)<ppyczko@apsl.net>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class RepairOrder(models.Model):
+    _inherit = "repair.order"
+
+    product_location_dest_id = fields.Many2one(
+        "stock.location",
+        "Product Destination Location",
+        compute="_compute_product_location_dest_id",
+        store=True,
+        readonly=False,
+        required=True,
+        precompute=True,
+        index=True,
+        check_company=True,
+        help="This is the location where the repaired product will be stored.",
+    )
+
+    @api.depends("picking_type_id")
+    def _compute_product_location_dest_id(self):
+        for repair in self:
+            repair.product_location_dest_id = (
+                repair.picking_type_id.default_product_location_dest_id
+            )
+
+    def action_repair_done(self):
+        res = super().action_repair_done()
+
+        for record in self:
+            record.move_id.location_dest_id = record.product_location_dest_id.id
+        return res
```

Comment:
@ppyczko This location update seems less than ideal. You are changing the destination location of the repaired product **after** its completion. If you look at the parent method, the default inheritability of the method that creates the stock move is less than ideal. (And what about the quantities of the product that should be set in the `stock.quant` table? My guess would be that this is not updated properly.
I think you have 2 options here:
1. Pass a context value in this method which should be detected the create method of the stock move, then prior to creation of the product stock move update the destination location.
2. Change the implementation entirely by creating an additional stock move which will transfer the product to a different location after completing the repair. (I believe this will also make the traceability of the product clearer as the repair and the move are split and not done in the same operation.

Reply:
Thank you for the detailed feedback @PieterPaulussen. I went with option 2, of creating an additional stock move to transfer the repaired product, as it makes sense to have this move split from the repair move itself.

Could you review again please? Thanks!

Reply:
What do you think about it @LoisRForgeFlow?